pinot-common/src/main/java/org/apache/pinot/common/utils/ScalingThreadPoolExecutor.java

```diff
@@ -100,29 +100,42 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
    */
   static class ScalingQueue<E> extends LinkedBlockingQueue<E> {
 
-    private ThreadPoolExecutor _executor;
+    AtomicInteger _currentIdleThreadCount = new AtomicInteger(0);
 
     // Creates a queue of size Integer.MAX_SIZE
     public ScalingQueue() {
       super();
     }
 
-    // Sets the executor this queue belongs to
-    public void setThreadPoolExecutor(ThreadPoolExecutor executor) {
-      _executor = executor;
+    @Override
+    @Nonnull
```

Comment:
We don't usually annotate `@Nonnull` but assume everything is non-null if not annotated. Can you remove this and annotate `poll()` return as `@Nullable` instead?